dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -383,14 +383,33 @@ public Workflow updateWorkflow(@ApiParam(hidden = true) @Auth User user,
         }
 
         c.updateInfo(workflow);
-
         Workflow result = workflowDAO.findById(workflowId);
         Helper.checkEntry(result);
 
         return result;
 
     }
 
+    @PUT
+    @Timed
+    @UnitOfWork
+    @Path("/{workflowId}/resetVersionPaths")
+    @ApiOperation(value = "Change the workflow paths", notes = "Workflow version correspond to each row of the versions table listing all information for a workflow", response = WorkflowVersion.class, responseContainer = "List")
+    public Set<WorkflowVersion> updateWorkflowPath(@ApiParam(hidden = true) @Auth User user,
+                                   @ApiParam(value = "Workflow to modify.", required = true) @PathParam("workflowId") Long workflowId,
+                                   @ApiParam(value = "Workflow with updated information", required = true) Workflow workflow){
+
```

Comment:
You'll need to add 
`Helper.checkEntry(workflow);
 Helper.checkUser(user, workflow);` 
ala the above resource in order to make sure people are editing entries that they actually own (and that the workflow exists).